src/ProjectTemplates/Microsoft.Extensions.AI.Templates/Microsoft.Extensions.AI.Templates.csproj

```diff
@@ -24,55 +24,63 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <!--
-      Reference the GenerateTemplateContent project so that template content gets generated prior
-      to building this project.
-    -->
-    <ProjectReference
-      Include="..\GenerateTemplateContent\GenerateTemplateContent.csproj"
-      ReferenceOutputAssembly="false"
-      PrivateAssets="all" />
+    <TemplatePackageVersion Include="Microsoft.Extensions.AI"
```

Comment:
IIUC you don't want these to behave like ProjectReferences WRT to restore (transitive dependencies, resolving assets, selecting best TargetFramework, etc).  If that's the case, then it's OK to use an item different than ProjectReference since you'd just end up fighting with ResolveReferences to turn all that stuff off.

I would recommend that you make the path the identity though, and then the name can just be a `%(FileName)` transform of that.

Make the authored items be different for Project vs Package, and make the act of resolving the project items create the common package ones (with Id and Version).